src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7146,7 +7143,7 @@ private BoundExpression ConvertToArrayIndex(BoundExpression index, SyntaxNode no
         private BoundExpression TryImplicitConversionToArrayIndex(BoundExpression expr, WellKnownType wellKnownType, SyntaxNode node, DiagnosticBag diagnostics)
         {
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            TypeSymbol type = GetWellKnownType(wellKnownType, ref useSiteDiagnostics);
+            TypeSymbol type = GetWellKnownTypeWithoutRecordingUsage(wellKnownType, ref useSiteDiagnostics);
```

Comment:
>GetWellKnownTypeWithoutRecordingUsage [](start = 30, length = 37)

Everywhere else we've been using an optional parameter. Consider removing this method and just using that parameter.

Reply:
> Everywhere else we've been using an optional parameter. Consider removing this method and just using that parameter.

This method never records usage, I see no point in adding any useless optional parameter.

---
In reply to: [351358299](https://github.com/dotnet/roslyn/pull/40010#discussion_r351358299) [](ancestors = 351358299)